pkg/liqoctl/offload/handler.go

```diff
@@ -72,12 +77,59 @@ func (o *Options) Run(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, o.Timeout)
 	defer cancel()
 
+	var offloadNamespaces []string
+	offloadNamespaces = append(offloadNamespaces, o.Namespaces...)
+
+	if o.LabelSelector != "" {
+		// Parse the label selector
+		selector, err := labels.Parse(o.LabelSelector)
+		if err != nil && o.LabelSelector != "" {
+			return fmt.Errorf("invalid label selector: %w", err)
+		}
+
+		// List namespaces
+		var selectedNsList corev1.NamespaceList
+		if err := o.CRClient.List(ctx, &selectedNsList, &client.ListOptions{LabelSelector: selector}); err != nil {
+			return fmt.Errorf("cannot list namespace objects: %w", err)
+		}
+
+		for _, ns := range selectedNsList.Items {
+			if !slices.Contains(offloadNamespaces, ns.Name) {
+				offloadNamespaces = append(offloadNamespaces, ns.Name)
+			}
+		}
+	}
+
+	if len(offloadNamespaces) == 0 {
+		o.Printer.Info.Println("No namespaces can be offloaded")
+		return nil
+	}
+
+	// Offload each namespace
+	var errors []error
+	for _, ns := range offloadNamespaces {
+		if err := o.runOffload(ctx, ns); err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	if len(errors) > 0 {
+		return fmt.Errorf("encountered %d errors during offloading, error details: %v", len(errors), errors)
+	}
+
+	return nil
+}
+
+// runOffload encapsulates the core logic of the offload namespace command.
+func (o *Options) runOffload(ctx context.Context, namespace string) error {
 	// Output the NamespaceOffloading resource, instead of applying it.
 	if o.OutputFormat != "" {
 		o.Printer.CheckErr(o.output())
```

Comment:
I believe this doesn't work, you are not providing the name of the namespace to the output function:
```suggestion
		o.Printer.CheckErr(o.output(namespace))
```